core/appmodule/module.go

```diff
@@ -64,3 +65,14 @@ type HasPrecommit interface {
 	appmodule.AppModule
 	Precommit(context.Context) error
 }
+
+// HasName is an extension interface that must return the appmodule.AppModule's Name.
+type HasName interface {
```

Comment:
Should we add an alias for this in appmodulev2? I remember it was considered as a legacy thing as we can get the module name as a depinject.ModuleKey, but the module manager doesn't really cares and is still very dependent on Name().

Cc @aaronc

Reply:
Maybe we should make in a follow up the runtime/v2 module manager a bit less dependent on this?